### PR TITLE
Test2.0 BE-021(Anton Rud)

### PR DESCRIPTION
**Pull request by rudik-byte**
Added TestContainers
Writen 2 service and 2 controller test-classes
100% coverage
Docker work correctly

### CODE REVIEW

**Review by Hanna-Kalashnyk — DISMISSED** [review]

File: src/test/java/com/exadel/discount/controller/CountryControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+@Transactional
+@ActiveProfiles("test")
+public class CountryControllerTest extends AbstractIT {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Test
+    @WithMockUser(username = "admin", roles = {"USER", "ADMIN"})
+    public void getAllCountriesTest() throws Exception {
```
Comment (mikevoloshyn) [inline src/test/java/com/exadel/discount/controller/CountryControllerTest.java]: The test word should be at the beginning of the method name. Like
"testGetAllCountriesForAdmin" or "testGetAllCountries" if its not connected with user role.

File: src/test/java/com/exadel/discount/controller/TagControllerTests.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    @Autowired
+    private MockMvc mockMvc;
+
+
+    @Test
+    @WithMockUser(username = "admin", roles = {"USER", "ADMIN"})
+    public void getAllTagsTest() throws Exception {
+        mockMvc.perform(MockMvcRequestBuilders.get("/tags"))
+                .andExpect(MockMvcResultMatchers.status().isOk())
+                .andExpect(MockMvcResultMatchers.status().is2xxSuccessful());
```
Comment (mikevoloshyn) [inline src/test/java/com/exadel/discount/controller/TagControllerTests.java]: this test only tests HTTP status. Why it does not check the response body? Even If it should be empty you should check that.

File: src/test/java/com/exadel/discount/service/CountryServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+
+    @Test
+    public void testSave() {
+        var result = countryService.save(input);
+        Assertions.assertNotNull(result);
+    }
+
+    @Test
+    public void testFindALl() {
```
Comment (mikevoloshyn) [inline src/test/java/com/exadel/discount/service/CountryServiceTest.java]: typo

File: src/test/java/com/exadel/discount/service/CountryServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        when(countryRepository.findByName(anyString())).thenReturn(Optional.of(country));
+        CountryServiceImpl c = Mockito.mock(CountryServiceImpl.class);
+        CountryDTO actual = c.findByName(anyString());
+
+        Assertions.assertEquals(actual,input);
+
+    }
+
+    @Test
+    public void testExceptionFindByName(){
```
Comment (mikevoloshyn) [inline src/test/java/com/exadel/discount/service/CountryServiceTest.java]: rename to 
testFindByNameThrowsNotFoundException

**Review by mikevoloshyn — DISMISSED** [review]

File: src/test/java/com/exadel/discount/controller/CountryControllerTest.java (hunk 1 above)
Comment (rudik-byte) [reply src/test/java/com/exadel/discount/controller/CountryControllerTest.java]: done

File: src/test/java/com/exadel/discount/service/CountryServiceTest.java (hunk 3 above)
Comment (rudik-byte) [reply src/test/java/com/exadel/discount/service/CountryServiceTest.java]: done

File: src/test/java/com/exadel/discount/service/CountryServiceTest.java (hunk 4 above)
Comment (rudik-byte) [reply src/test/java/com/exadel/discount/service/CountryServiceTest.java]: done

File: src/test/java/com/exadel/discount/controller/TagControllerTests.java (hunk 2 above)
Comment (rudik-byte) [reply src/test/java/com/exadel/discount/controller/TagControllerTests.java]: done

**Review by ghost — DISMISSED** [review]

**Review by Hanna-Kalashnyk — DISMISSED** [review]

**Review by el-dub — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**